metadata_backend/api/handlers.py

```diff
@@ -523,24 +503,71 @@ def _check_patch_folder(self, patch_ops: Any) -> None:
                             raise web.HTTPBadRequest(reason=reason)
 
     async def get_folders(self, req: Request) -> Response:
-        """Get all possible object folders from database.
+        """Get a set of folders owned by the user based on pagination values.
 
         :param req: GET Request
         :returns: JSON list of folders available for the user
         """
+
+        def get_page_param(param_name: str, default: int) -> int:
+            """Handle page parameter value extracting."""
+            try:
+                param = int(req.query.get(param_name, default))
+            except ValueError:
+                reason = f"{param_name} parameter must be a number, now it is {req.query.get(param_name)}"
+                LOG.error(reason)
+                raise web.HTTPBadRequest(reason=reason)
+            if param < 1:
+                reason = f"{param_name} parameter must be over 0"
+                LOG.error(reason)
+                raise web.HTTPBadRequest(reason=reason)
+            return param
+
+        page = get_page_param("page", 1)
+        per_page = get_page_param("per_page", 5)
         db_client = req.app["db_client"]
 
         user_operator = UserOperator(db_client)
-
         current_user = get_session(req)["user_info"]
         user = await user_operator.read_user(current_user)
 
-        operator = FolderOperator(db_client)
-        folders = await operator.query_folders({"folderId": {"$in": user["folders"]}})
+        folder_query = {"folderId": {"$in": user["folders"]}}
+        # Check if only published or draft folders are requestsed
+        if "published" in req.query:
+            pub_param = req.query.get("published", "").lower()
+            if pub_param == "true":
```

Comment:
we can check pub_param in list of true or false then in the query we convert pub_param to boolean to simplify this check

Reply:
Good point. So like this?
```suggestion
            if "published" in req.query:
                pub_param = req.query.get("published", "").title()
                if pub_param in ["True", "False"]:
                    folder_query["published"] = {"$eq": eval(pub_param)}
```

Reply:
`pub_param in ["True", "False"]:` maybe lowercase/standardise the value as we don't know what the input will look like

Reply:
`.title()` capitalizes first letter of a string so the idea being that inputs like `"false"`, `"FALSE"` or `"faLse"` would all become `"False"`

Reply:
i see - well ok